gtr/gtr.go

```diff
@@ -73,10 +73,28 @@ type Package struct {
 // property with the given key already exists, its old value will be
 // overwritten with the given value.
 func (p *Package) SetProperty(key, value string) {
-	if p.Properties == nil {
-		p.Properties = make(map[string]string)
+	// TODO(jstemmer): Delete this method in the next major release.
+	// Delete all the properties whose name is the specified key,
+	// then add the specieid key-value property.
+	i := 0
+	for _, prop := range p.Properties {
+		if key == prop.Name {
+			p.Properties[i] = prop
+			i++
+		}
 	}
-	p.Properties[key] = value
+	p.Properties = p.Properties[:i]
+	p.AddProperty(key, value)
+}
+
+// AddProperty appends a name/value property in the current package.
+func (p *Package) AddProperty(name, value string) {
+	p.Properties = append(p.Properties, Property{Name: name, Value: value})
+}
+
+// Property is a name/value property.
+type Property struct {
```

Comment:
I could have reused junit.Property here, but that would have introduced a gtr->junit dependency that doesn't currently exist.

Reply:
It also would've introduced an import cycle, so I think this is the right approach.